### DISCUSSION TITLE
Line break tags <br />in Markdown are not being processed with `remark` and `remark-html`

### DISCUSSION THREAD

**Discussion by Arash-11** [remarkjs/remark#858] [Q&A] [answered]
When I try converting Markdown to HTML using `remark` and `remark-html`, the `<br />` tags are not being inserted in the final HTML that's being processed.

For instance, when I run the following:
```
import remark from 'remark';
import remarkHtml from 'remark-html';

const processedContent = await remark().use(remarkHtml).process(
  '## Hello <br /> world'
);
```

`processedContent` outputs `'<h2>Hello  world</h2>\n'`. Is there something I'm missing here?

**Comment by ChristianMurphy** [comment, marked as answer]
`remark` is just markdown, `remark-html` converts just markdown to HTML.
HTML is not a part of Markdown, it can be included in Markdown, but it is essentially an embedded document.

This is why the readme for `remark-html` notes:

> Note that raw HTML in Markdown cannot be sanitized, so it’s removed. A schema can still be used to allow certain values from integrations though. To support HTML in Markdown, use rehype-raw.

https://github.com/remarkjs/remark-html#optionssanitize

this can be resolved using hints from the other note in the `remark-html` readme

> warning This package essentially packs remark-rehype and rehype-stringify, and although it does support some customisation, it isn’t very pluggable. It’s probably smarter to use remark-rehype directly and benefit from the rehype ecosystem.

https://github.com/remarkjs/remark-html#remark-html

Using `remark-rehype` to handle this, add `allowDangerousHtml` to `remark-rehype` to keep the embedded HTML documents, use `rehype-raw` to parse the embedded documents into the HTML AST (HAST), and apply `rehype-santize` to clear potentially dangerous content.

Put together it would look something like: https://codesandbox.io/s/remark-rehype-with-raw-html-xrpp8?file=/src/index.js

**Reply by wooorm** [reply]
See also https://unifiedjs.com/learn/recipe/remark-html/!

**Reply by Arash-11** [reply]
I fixed it, thank you for the info!

**Reply by maltejur** [reply]
@ChristianMurphy how would I do this with `react-markdown`? I already tried this

```jsx
<ReactMarkdown
  remarkPlugins={[
    remarkParse,
    [remarkRehype, { allowDangerousHtml: true }],
    rehypeRaw,
    rehypeSanitize,
    [rehypeStringify, { allowDangerousHtml: true }],
  ]}
>
  {`# Title

Text<br />More Text`}
</ReactMarkdown>
```

But there the markdown itself doesn't get rendered anymore.

**Reply by ChristianMurphy** [reply]
@maltejur see the `react-markdown` readme on including HTML https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown

**Reply by maltejur** [reply]
Thank you, that worked. My error was that I didn't realize that there was a difference between `remarkPlugins` and `rehypePlugins` and I didn't see the appendix.
